### ISSUE TITLE
is/assertType functions throw error when interface has Date field

### ISSUE THREAD

**Issue by watson28** [woutervh-/typescript-is#17]
I have an interface which has a date field and when I try to assert an object against this interface the library throws the following error

Failed to transform node at:  at transformNodeAndChildren (/xxx/node_modules/typescript-is/lib/transform-inline/transformer.js:27

```
interface MyInterface {
  id: string;
  createdAt: Date;
  modifiedAt: Date;
}
```

Does the library doesnt support interfaces with `Date` fields?

**Comment by woutervh-** [comment]
Hi @watson28 

Thanks for reporting the issue.

Validating classes (which also includes Date) is not supported. In issue #16 a similar request came in for the RegExp class. The solution I'm looking at right now is to add an option to ignore classes and methods rather than throw an error.

Would that be an acceptable alternative?

**Comment by watson28** [comment]
@woutervh- thanks for your quick response. I think that's a good workaround. 
its really bad the library doesn't support classes. is there any plan for supporting classes in the future?

**Comment by woutervh-** [comment]
hi @watson28 

The workaround has been released (see my comment in #16 for details).

Supporting classes is not something that I see as a high priority for this library because it's focused on type guarding serializable data. RegExp, Date, etc. are all created at runtime, so somewhere in your code where you are creating an instance of these classes you should already have the type available and be type safe. Reading files or fetching data from an HTTP server are examples where you can't be sure you have the right type, and that's what this library is mainly for.

Adding support for classes is something that could be done in the long run, but I think it would be tricky, because I'd have to make sure that the class you are validating is in fact imported, and I'd have to figure out the name of the class in your runtime. I'm not sure this possible.

If you have a solution to this problem, feel free to contribute 👍

**Comment by woutervh-** [comment]
Closing for house keeping.
